server/pulp/server/db/model/dispatch.py

```diff
@@ -221,6 +222,14 @@ def as_dict(self):
             'total_run_count': self.total_run_count,
         }
 
+        for key, value in dict_repr.iteritems():
+            try:
+                dict_repr[key] = value.serializer(value).data
+            except AttributeError:
+                continue
```

Comment:
Yeah, that could use a comment. I'll add something like this:

The idea is that any value that has a `.serializer` attribute has been converted to Mongoengine. Because of this, it does not inherit from `dict` like the old models and will need to use that serializer. If value is not a mongoengine object it will not have `.serializer` and will raise an attribute error. In this case, don't try to serialize it and check the next item.